File: backend/src/main/java/com/scim/service/AuthService.java
package com.scim.service;

import com.scim.dto.auth.request.LoginRequest;
import com.scim.dto.auth.request.RefreshTokenRequest;
import com.scim.dto.auth.response.AuthResponse;

public interface AuthService {
    AuthResponse login(LoginRequest request);
    AuthResponse refresh(RefreshTokenRequest request);
    void logout(String refreshToken);
}

File: frontend/src/services/authService.ts
import { apiClient } from '@/config/apiClient';
import type { ApiResponse, LoginRequest, LoginResponse } from '@/types';

export const authService = {
  login: (data: LoginRequest): Promise<ApiResponse<LoginResponse>> => {
    return apiClient.post('/auth/login', data);
  },

  refresh: (refreshToken: string): Promise<ApiResponse<LoginResponse>> => {
    return apiClient.post('/auth/refresh', { refreshToken });
  },

  logout: (refreshToken: string): Promise<ApiResponse<void>> => {
    return apiClient.post('/auth/logout', { refreshToken });
  },
};
